include/envoy/common/token_bucket.h

```diff
@@ -21,6 +21,11 @@ class TokenBucket {
    * @return true if bucket is not empty, otherwise it returns false.
    */
   virtual bool consume(uint64_t tokens = 1) PURE;
+
+  /**
+   * @return returns the time in milli seconds indicating when the next token would be available.
```

Comment:
It's not quite that, right? It's some approximation. If there is a token available, it correctly returns 0, otherwise it will give you an upper bound on the amount of time until a next token is available based on the implementation below.

Do you think this approximation is reasonable, or do you want to actually compute from the clock in the token bucket something closer?

I think it becomes a much bigger issue for very low rate limits, e.g. 0.001. If it's once a second, for the purpose of xDS, it doesn't matter as much..

Reply:
I will change the wording, but I think this approximation should be good enough for xDS type of use cases. I will add that to the documentation